indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -844,6 +858,48 @@ private void removeWorker(final Worker worker)
     }
   }
 
+  private void scheduleTasksCleanupForWorker(final String worker, final List<String> tasksToFail)
+  {
+    removedWorkerCleanups.put(
+        worker, exec.schedule(
+            new Runnable()
+            {
+              @Override
+              public void run()
+              {
+                log.info("Running scheduled cleanup for Worker[%s]", worker);
+                for (String assignedTask : tasksToFail) {
```

Comment:
In the presence of very spotty networking, the host could come and go. In such a case it would make sense to me to have a "last seen on" check somehow so that the task could be skipped on this round of purging if it was seen within 5 minutes.